src/test/java/igrad/model/module/SemesterTest.java

```diff
@@ -1,5 +1,28 @@
 package igrad.model.module;
 
-// TODO: (Teri) Implement all the relevant test methods
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
 public class SemesterTest {
+
+    @Test
+    public void constructor_invalidSemester_throwsIllegalArgumentException() {
+        String invalidSemester = "Y1S3";
+        assertThrows(IllegalArgumentException.class, () -> new Semester(invalidSemester));
+    }
+
+    @Test
+    public void isValidSemester() {
+        // invalid semester
+        assertFalse(Semester.isValidSemester("Y1S3"));
```

Comment:
good

Reply:
but prob want to rearrange abit, the one below and this one should swap places

Reply:
also, you might want to add on to your existing comments like; `// invalid semester (Sem 3 is not a valid sem)` to  explain why it is invalid